arc/database/batch_processor.py

```
@@ -0,0 +1,386 @@
+        finally:
+            self._failure_batch.clear()
+    
+    async def _retry_operation(self, operation: Callable):
+        """Retry an operation with exponential backoff."""
+        for attempt in range(self.config.max_retries):
+            try:
+                return await operation()
+            except Exception as e:
+                if attempt == self.config.max_retries - 1:
+                    raise
+                
```

Ruff (0.11.9)

Blank line contains whitespace

Remove whitespace from blank line

(W293)